### PR TITLE
[4.0] WebAsset for plugins

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes
This replaces htmlhelper to webaset for plugins

Updated plugins:
```
quickicon/overridecheck
quickicon/extensionupdate
quickicon/privacycheck
quickicon/joomlaupdate
system/webauthn
system/stats
editors-xtd/readmore
editors/codemirror
editors/tinymce
installer/webinstaller
```

### Testing Instructions
Apply patch.
Review all updated plugins

### Expected result
All works

### Actual result
All works

### Documentation Changes Required
nope 

ref #22435

### CODE REVIEW

File: plugins/editors-xtd/readmore/readmore.php [hunk 1]
```diff
@@ -41,10 +40,13 @@ class PlgButtonReadmore extends CMSPlugin
 	 */
 	public function onDisplay($name)
 	{
-		HTMLHelper::_('script', 'com_content/admin-article-readmore.min.js', array('version' => 'auto', 'relative' => true));
+		/** @var Joomla\CMS\Document\HtmlDocument $doc */
+		$doc = Factory::getApplication()->getDocument();
```
Comment (wilsonge) [inline plugins/editors-xtd/readmore/readmore.php]: Use the application property in the plugin please (in all the places you're doing `\JFactory::getApplication`- if that requires the property being added - then add it

Comment (Fedik) [reply plugins/editors-xtd/readmore/readmore.php]: changed

File: plugins/quickicon/joomlaupdate/src/Extension/Joomlaupdate.php [hunk 2]
```diff
@@ -123,8 +122,8 @@ public function getCoreUpdateNotification(QuickIconsEvent $event)
 			]
 		);
 
-		HTMLHelper::_('behavior.core');
-		HTMLHelper::_('script', 'plg_quickicon_joomlaupdate/jupdatecheck.min.js', array('version' => 'auto', 'relative' => true));
+		$this->app->getDocument()->getWebAssetManager()
```
Comment (SharkyKZ) [inline plugins/quickicon/joomlaupdate/src/Extension/Joomlaupdate.php]:
```suggestion
		$this->document->getWebAssetManager()
```

File: plugins/editors/codemirror/codemirror.php [hunk 3]
```diff
@@ -11,7 +11,6 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\Factory;
```
Comment (Quy) [inline plugins/editors/codemirror/codemirror.php]:
```suggestion
```

File: plugins/system/stats/layouts/field/data.php [hunk 4]
```diff
@@ -9,10 +9,11 @@
 
 defined('JPATH_BASE') or die;
 
-use Joomla\CMS\HTML\HTMLHelper;
 use Joomla\CMS\Language\Text;
```
Comment (Quy) [inline plugins/system/stats/layouts/field/data.php:13]:
```suggestion
use Joomla\CMS\Factory;
use Joomla\CMS\Language\Text;
```

**Comment by wilsonge** [comment]
LGTM! Thanks!

**Pull request merged**